[Cleanup] Implementing Warning Banner For Payments If Amout Is Negative

Civolilah opened the pull request:
@beganovich @turbo124 The PR implements a warning banner that appears when a negative amount is entered. Screenshot:

<img width="1261" alt="Screenshot 2024-11-20 at 19 01 40" src="https://github.com/user-attachments/assets/63bd1752-875d-4e94-ba17-c3a154e98c90">

Let me know your thoughts.

turbo124 commented:
The banner appears after the user has already saved the payment. 

The banner should appear under IF the amount is negative using either debounce/onBlur After the payment is saved, this warning is too late.

Civolilah commented:
> The banner appears after the user has already saved the payment.
> 
> The banner should appear under IF the amount is negative using either debounce/onBlur After the payment is saved, this warning is too late.

@turbo124 Okay, so I've changed the approach. Instead of using an onBlur event to update the field's value, I've implemented a debounce override. Now, when the value changes on the Create page, if it's negative, the banner will appear after a 300ms delay. Let me know your thoughts.
